### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by Ihmoda**

### CODE REVIEW

File: curriculum/js2/5.js [hunk 1]
```diff
@@ -4,8 +4,16 @@
  * @returns {number}
  */
 
-const solution = (a) => {
-  return 0;
+const solution = (a, idx = 1, largest = a[0]) => {
+  if (idx >= a.length) {
+    return a.length === 0 ? '' : largest;
```
[the remark this answers is not in the document]
Comment (Ihmoda) [reply curriculum/js2/5.js]: Please see comment on 4_js. Returning string to satisfy tests.

File: curriculum/js2/5.js [hunk 2]
```diff
@@ -4,8 +4,16 @@
  * @returns {number}
  */
 
-const solution = (a) => {
-  return 0;
+const solution = (a, idx = 1, largest = a[0]) => {
+  if (idx >= a.length) {
+    return a.length === 0 ? 0 : largest;
```
Comment (seemcat) [inline curriculum/js2/5.js]: same comment as my last. this ternary operator is unnecessary to make the code work, no?

**Review by seemcat — CHANGES_REQUESTED** [review]

File: curriculum/js2/5.js (hunk 2 above)
Comment (Ihmoda) [reply curriculum/js2/5.js]: The previous one did not require a ternary, but I believe this one does need it to handle the empty array case. In the prior example, sum was initialized to zero. In this example a[0] will evaluate to undefined and we want to return zero in that case.

**Comment by Ihmoda** [comment]
Original Problem: 

/**
 * Given an array, return the largest element
 * @param {array} a
 * @returns {number}
 */

const solution = (a) => {
  return 0;
};

module.exports = {
  solution,
};

File: curriculum/js2/5.js (hunk 2 above)
Comment (songz) [inline curriculum/js2/5.js]: `return largest || -1` Usually I use -1 for invalid input for functions that should return a number.

**Review by songz — APPROVED** [review]

**Pull request merged**